server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -148,4 +155,9 @@ public MailAddress getMailAddressFor(Username username) throws UsersRepositoryEx
             throw new UsersRepositoryException("Failed to compute mail address associated with the user", e);
         }
     }
+
+    private boolean assertLocalPartValid(Username username) {
```

Comment:
you named the method `assert...` I guess it could throw

Reply:
You are right. `isLocalPartValid` would be better?

Reply:
you are right. `isLocalPartValid` sounds better?

Reply:
or you can move the if check inside the method an keep the name `assertLocalPartValid()`. It's up to you